docs/asyncio_imports.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Importing code with asyncio"
+---
+
+Determining when it is safe to import code when using asyncio can be tricky because two constraints need to be considered:
+
+- Importing code can do blocking I/O to load the files from the disk
+- Importing code in [cpython is not thread-safe](https://github.com/python/cpython/issues/83065)
+
+## Top-level imports
+
+Suppose your imports happen at the top-level (nearly all code at indentation level 0). Home Assistant will import your code before the event loop starts or import it in the import executor when your integration is loaded. In this case, you likely do not need to consider whether your imports are safe.
```

Comment:
I think the correct term is "module level" since it mentions indentation level 0?
Should we explain there needs to a chain of imports starting from a package's `__init__.py`?